contracts/tokenomics/treasury/src/state.rs

```diff
@@ -21,10 +22,23 @@ pub struct Config {
     /// Address of the security DAO contract
     pub security_dao_address: Addr,
 
-    // Denomintator used int the vesting release function
+    // Denomintator used in the vesting release function
     pub vesting_denominator: u128,
 }
 
+impl Config {
+    pub fn validate(&self) -> Result<(), ContractError> {
+        if (self.distribution_rate > Decimal::one()) || (self.distribution_rate < Decimal::zero()) {
+            return Err(ContractError::InvalidDistributionRate {});
+        }
+
+        if self.vesting_denominator == 0 {
+            return Err(ContractError::InvalidVestingDenominator {});
+        }
+        Ok(())
```

Comment:
maybe add check for `min_period > 0`?

Reply:
in the task it also says to validate denom, but I'm not sure how

Reply:
neither do I:

> idk what kind of validation denom and min_period require and what's wrong with the vesting_denominator validation, so I kept the original validation but moved it to a Config struct method and added tests for it. If you have an idea please let me know.

yeah, min_period > 0 seems to be the only thing we can validate